### ISSUE TITLE
Make It Modular

### ISSUE THREAD

**Issue by Kiwilicious** [workshopper/learnyounode#497]
I'm attempting the Make It Modular challenge. 
This is my module exports file.
```
module.exports = function(target, ext, callback) {
    var fs = require('fs'),
        path = require('path')
        data = [];

    fs.readdir(target, 'utf8', (err, files) => {
        if (err) return callback(err);
        files.forEach(file => {
            var fileExt = path.extname(file);
            if (fileExt === ext) data.push(file);
        })
        return callback(null, data);
    })
};
```

This is my main file which calls upon the module file.
```
var myModule = require('./MY_MODULE')
    target = process.argv[2],
    ext = '.' + process.argv[3];

function callback(err, data) {
    if (err) throw err;
    data.forEach(file => console.log(file));
}

myModule(target, ext, callback);
```

This is the result. I'm confused as to why it says that I'm not passing back the error from fs.readdir(). I looked over it and it seems to be alright. 
```
Your submission results compared to the expected:

                 ACTUAL                                 EXPECTED
────────────────────────────────────────────────────────────────────────────────

   "CHANGELOG.md"                      ==    "CHANGELOG.md"
   "LICENCE.md"                        ==    "LICENCE.md"
   "README.md"                         ==    "README.md"
   ""                                  ==    ""

────────────────────────────────────────────────────────────────────────────────

 ✓

 Submission results match expected

 ✓

 Additional module file exports a single function

 ✓

 Additional module file exports a function that takes 3 arguments

 ✗

 Your additional module file [MY_MODULE.js] does not appear to pass back an
 error received from fs.readdir(). Use the following idiomatic Node.js
 pattern inside your callback to fs.readdir(): if (err) return
 callback(err)

 # FAIL Your solution to MAKE IT MODULAR didn't pass. Try again!

─────────────────────────────────────────────────────────────────────────────
```

**Comment by Chrinkus** [comment]
Its the throw statement. The workshop is probably looking for the error to be reported in the console.

**Comment by martinheidegger** [comment]
Closed by #502
